src/rcclScalarAllGatherKernels.h

```diff
@@ -12,45 +12,111 @@ All rights reserved.
  * This file contains implementation of kernels used by rcclAllGather
  *
  */
+template<class T>
+__device__ bool is_aligned(const void * ptr) noexcept {
+    auto iptr = reinterpret_cast<std::uintptr_t>(ptr);
+    return !(iptr % alignof(T));
+}
 
 //! @brief Definition of RcclKernelScalarAllGather
 //! Gather data from all gpus and store to current gpu destination buffer
 template <typename DataType_t>
 __global__ void RcclKernelScalarAllGather(RingNode_t* pcurr_track, int rank,
-                                          int count) {
-    int tx = threadIdx.x;
-    int bx = blockIdx.x;
-    int tid = tx + bx * knum_vectors_per_workgroup;
-
+        int count) {
+    int tid = threadIdx.x;
+    RingNode_t* pnext_track = pcurr_track->next_gpu;
+    bool use_64b_copy = 0;
+    int packed_count = count * sizeof(DataType_t) / sizeof(unsigned long long);
```

Comment:
You can add `VectorType_t` to templates and find the packed count as `count / (sizeof(VectorType_t) / sizeof(DataType_t))`